src/components/dashboard/main-content/course-enrollments/data/hooks.js

```diff
@@ -422,51 +422,52 @@ export const useUpdateCourseEnrollmentStatus = ({ enterpriseCustomer }) => {
  */
-export function useGroupMembershipAssignments() {
+export function useGroupsAlert() {
   const { data: enterpriseCustomer } = useEnterpriseCustomer();
-  const { data: enterpriseGroupMemberships } = useEnterpriseGroupMemberships();
-  const [shouldShowNewGroupMembershipAlert, setShouldShowNewGroupMembershipAlert] = useState(false);
-  const [groupMembershipsToDismiss, setGroupMembershipsToDismiss] = useState([]);
-  const checkIsGroupAssignmentDismissed = useCallback(() => {
-    enterpriseGroupMemberships.forEach(membership => {
-      const isDismissedGroupAssignment = global.localStorage.getItem(`${HAS_USER_DISMISSED_NEW_GROUP_ASSIGNMENT_ALERT}-${membership.groupUuid}`);
-      if (isDismissedGroupAssignment === null) {
-        setGroupMembershipsToDismiss(prevState => [...prevState, membership]);
-        setShouldShowNewGroupMembershipAlert(true);
+  const { data: { redeemablePolicies } } = useRedeemablePolicies();
+  const [shouldShowNewGroupAlert, setShouldShowNewGroupAlert] = useState(false);
+  const [groupAssociationUUIDs, setGroupAssociationUUIDs] = useState([]);
+
+  const checkIsGroupDismissed = useCallback(() => {
+    redeemablePolicies.forEach(policy => {
+      // Getting the first index because each policy only has 1 group association
+      const groupAssociationUUID = policy.groupAssociations[0];
+      if (groupAssociationUUID) {
+        const isDismissedGroup = global.localStorage.getItem(`${HAS_USER_DISMISSED_NEW_GROUP_ALERT}-${groupAssociationUUID}`);
+        if (isDismissedGroup === null) {
+          setGroupAssociationUUIDs(prevState => [...prevState, groupAssociationUUID]);
+          setShouldShowNewGroupAlert(true);
+        }
       }
     });
-  }, [enterpriseGroupMemberships]);
+  }, [redeemablePolicies]);
 
-  const handleAddNewGroupAssignmentToLocalStorage = () => {
-    groupMembershipsToDismiss.forEach(groupMembership => {
-      global.localStorage.setItem(`${HAS_USER_DISMISSED_NEW_GROUP_ASSIGNMENT_ALERT}-${groupMembership.groupUuid}`, true);
+  const handleAddNewGroupToLocalStorage = () => {
+    groupAssociationUUIDs.forEach(groupAssociationUUID => {
+      global.localStorage.setItem(`${HAS_USER_DISMISSED_NEW_GROUP_ALERT}-${groupAssociationUUID}`, true);
     });
-    setGroupMembershipsToDismiss([]);
-    setShouldShowNewGroupMembershipAlert(false);
-    checkIsGroupAssignmentDismissed();
+    setGroupAssociationUUIDs([]);
+    setShouldShowNewGroupAlert(false);
+    checkIsGroupDismissed();
   };
 
-  /**
-   * Parses the enterprise group memberships and checks if they have already been
-   * acknowledged (dismissed) by the learner.
-   */
   useEffect(() => {
-    checkIsGroupAssignmentDismissed();
-  }, [checkIsGroupAssignmentDismissed]);
+    checkIsGroupDismissed();
+  }, [checkIsGroupDismissed]);
 
   return {
-    shouldShowNewGroupMembershipAlert,
-    handleAddNewGroupAssignmentToLocalStorage,
+    shouldShowNewGroupAlert,
```

Comment:
nit: in the same vein as the suggestion to rename `handleAddNewGroupToLocalStorage ` -> `dismissGroupAssociationAlert ` (or similar), having this state variable include the word "association" may help to expand on the semantics for the alert (e.g., `showNewGroupAssocationAlert`).

For instance, `shouldShowNewGroupAlert` on its own _could_ be interpreted as a new group was added to the enterprise customer, but without the learner being associated to it as an example.